panel/core/esm.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import textwrap
+
+from typing import (
+    TYPE_CHECKING, Any, ClassVar, Dict, List, Literal, Mapping, Optional,
+    Tuple, Type,
+)
+
+import jinja2
+import param
+
+from param.parameterized import ParameterizedMetaclass
+
+from .io.datamodel import construct_data_model
+from .io.state import state
+from .models import ReactiveESM as _BkReactiveESM
+from .reactive import Reactive, ReactiveCustomBase, ReactiveMetaBase
+from .viewable import Layoutable, Viewable
+
+if TYPE_CHECKING:
+    from bokeh.document import Document
+    from bokeh.model import Model
+    from pyviz_comms import Comm
+
+
+class ReactiveESMMetaclass(ReactiveMetaBase):
+
+    def __init__(mcs, name: str, bases: Tuple[Type, ...], dict_: Mapping[str, Any]):
+        mcs.__original_doc__ = mcs.__doc__
+        ParameterizedMetaclass.__init__(mcs, name, bases, dict_)
+
+        # Create model with unique name
+        ReactiveMetaBase._name_counter[name] += 1
+        model_name = f'{name}{ReactiveMetaBase._name_counter[name]}'
+        mcs._data_model = construct_data_model(
+            mcs, name=model_name, ignore=list(Reactive.param)
+        )
+
+
+class ReactiveESM(ReactiveCustomBase, metaclass=ReactiveESMMetaclass):
+
+    _esm: ClassVar[str | os.PathLike] = ""
```

Comment:
I believe we lack the `_css` attribute that can also be watched. `AnyWidget` has this. I can use `_stylesheets` but why not use same name as AnyWidget and why not support string or  PurePath instead of a list of strings.

Reply:
> It does not support paths

It has always supported paths, hot reloading support now supported too.